metricbeat/scripts/docs_collector.py

```diff
@@ -0,0 +1,126 @@
+            continue
+
+        # Create directory for each module
+        os.mkdir(os.path.abspath("docs") + "/modules/" + module)
+
+        module_file = generated_note
+        module_file += "[[metricbeat-module-" + module + "]]\n"
+
+        with file(module_doc) as f:
+            module_file += f.read()
+
+
+        config_file = path + "/" + module + "/_beat/config.yml"
+
+        # Add example config file
+        if os.path.isfile(config_file) == True:
+
+            module_file += """
+
+=== Example Configuration
+
+The Apache module supports the standard configuration options which can be found
+here (add link). Below is an example of a configuration option:
+
+[source,yaml]
+----
+"""
+
+            # Load metricset yaml
+            with file(config_file) as f:
+                # Add 4 spaces for indentation in fron of each line
+                for line in f:
+                    if len(line.strip()) > 0:
+                        module_file += "  " + line
+                    else:
+                        module_file += line
+
+            module_file += "----\n\n"
+
+        # Add metricsets title as below each metricset adds its link
+        module_file += "=== MetricSets\n\n"
+
+        # Iterate over all metricsets
+        for metricset in os.listdir(base_dir + "/" + module):
+
+            metricset_docs = path + "/" + module + "/" + metricset + "/_beat/docs.asciidoc"
+
+            # Only check folders where fields.yml exists
+            if os.path.isfile(metricset_docs) == False:
+                continue
+
+            link_name = "metricbeat-metricset-" + module + "-" + metricset
+            link = "<<" + link_name + "," + metricset + ">>"
+            reference = "[[" + link_name + "]]"
+
+            module_file += "* " + link + "\n"
+
+
+            metricset_file = generated_note
+            # Add reference to metricset file
+            metricset_file += reference + "\n"
+
+            with file(metricset_docs) as f:
+                metricset_file += f.read()
+
+            metricset_file += """
+
+===== Fields
+
+A description of each field in the MetricSet can be found in the
+<<exported-fields-""" + module + """-""" + metricset + """,exported fields>> section
+
+"""
+
+            data_file = path + "/" + module + "/" + metricset + "/_beat/data.json"
+
+            # Add data.json example json document
+            if os.path.isfile(data_file) == True:
+                metricset_file += "Below is an example document generated by this metricset."
+                metricset_file += "\n\n"
+
+                metricset_file += "[source,json]\n"
+                metricset_file += "----\n"
+
+                with file(data_file) as f:
+                    metricset_file += f.read()
+
+                metricset_file += "----\n"
+
+            # Write metricset docs
+            f = open(os.path.abspath("docs") + "/modules/" + module + "/" + metricset + ".asciidoc", 'w')
```

Comment:
`with open() as f` would be better here to make sure the file is closed immediately after.